### PR TITLE
Misc. Zephyrization fixups

### PR DESCRIPTION
**Pull request by andyross**
Some changes needed by an imminent Mediatek Zephyrization PR, largely of the form "make this older code build with Zephyr while we untangle it from the legacy layers".

Most of these shouldn't be controversial, so I've left them in a long series for simplicity.  Let me know if there's something people want to split out for independent review.

In particular please look carefully at the Kconfig changes.  I'm 90%+ sure that this is pure cleanup that doesn't change any existing configs, but it's not possible for me to test (or even build) all platforms, so some help by the relevant maintainers would be appreciated.

### CODE REVIEW

**Comment by andyross** [comment]
Quick respin to fix a dropped file change.  And now I see that this breaks IPC3 fuzzing somewhere.  Will fix.

File: scripts/xtensa-build-zephyr.py [hunk 1]
```diff
@@ -787,10 +787,14 @@ def build_platforms():
 		_dict = dataclasses.asdict(platform_configs[platform])
 		platform_dict = { k:v for (k,v) in _dict.items() if _dict[k] is not None }
 
+		xtensa_tools_version = os.getenv("XTENSA_TOOLS_VERSION")
+		if not xtensa_tools_version:
+			xtesna_tools_version = platform_dict["XTENSA_TOOLS_VERSION"]
```
Comment (kv2019i) [inline scripts/xtensa-build-zephyr.py:792]: This is potentially a scary change. Whether hard coding the versions in this python file is good or bad, it's been in use for a long time and this has specifically been done to ensure all developers and the various build systems, use the same configuration to build. And more importantly, any change is done with a git commit that is tested with the same test process as other pull requests. If we now open a new getenv() path to modify the build, it's hard to ensure there are no CI builders out there, that have foobar XTENSA_TOOLS_VERSION contents and in worst-case, don't fail immediate but start generating slightly different binaries (and test runners test those).

So I'd like to make this more explicit. Use a different name for the environment variable, or at minimum print a prominent warning to log file that non-standard toolchain is used in the build. I don't think this is now printed at all.

File: src/lib/alloc.c [hunk 2]
```diff
@@ -533,7 +533,7 @@ static void free_block(void *ptr)
 #endif
 }
 
-#if CONFIG_TRACE
+#if defined(CONFIG_TRACE) && !defined(__ZEPHYR__)
```
[the remark this answers is not in the document]
Comment (kv2019i) [reply src/lib/alloc.c:536]: Ack, none of the current NXP/Intel targets link this with Zephyr (sof/zephyr/lib/alloc.c is used instead).

File: src/init/init.c [hunk 3]
```diff
@@ -287,7 +287,7 @@ static int primary_core_init(int argc, char *argv[], struct sof *sof)
 	interrupt_init(sof);
 #endif /* __ZEPHYR__ */
 
-#ifdef CONFIG_ZEPHYR_LOG
+#if defined(CONFIG_ZEPHYR_LOG) && !defined(CONFIG_LOG_MODE_MINIMAL)
```
Comment (kv2019i) [inline src/init/init.c:290]: Ack, this has been missed. Minimal support indeed a good addition!

File: zephyr/include/rtos/interrupt.h [hunk 4]
```diff
@@ -48,9 +49,7 @@ static inline void interrupt_unregister(uint32_t irq, const void *arg)
 
 static inline int interrupt_get_irq(unsigned int irq, const char *cascade)
 {
-#if defined(CONFIG_LIBRARY) || defined(CONFIG_ACE) || defined(CONFIG_CAVS) || \
-	defined(CONFIG_ZEPHYR_POSIX) || (defined(CONFIG_IMX) && !defined(CONFIG_IMX8M)) || \
-	defined(CONFIG_AMD)
+#if defined(CONFIG_LIBRARY) || defined(__ZEPHYR__) || defined(CONFIG_ARM64)
```
Comment (kv2019i) [inline zephyr/include/rtos/interrupt.h:52]: I don't think this is correct. I think IMX platforms have at least had a quirk here for irq support (note: "(defined(CONFIG_IMX) && !defined(CONFIG_IMX8M))"). I do agree this is getting way too complex, so at least for Intel platforms, "__ZEPHYR__" is sufficient as we've moved all platforms to Zephyr. I think for NXP, imx8m is a special case as it uses XTOS drivers and this is going away soon -> see https://github.com/thesofproject/sof/pull/9608 ).

File: posix/include/sof/lib/dma.h [hunk 5]
```diff
@@ -97,6 +97,9 @@ enum dma_cb_status {
 	DMA_CB_STATUS_END,
 };
 
+#define SOF_DMA_CB_STATUS_RELOAD	DMA_CB_STATUS_RELOAD
+#define SOF_DMA_CB_STATUS_END		DMA_CB_STATUS_END
+
```
Comment (kv2019i) [inline posix/include/sof/lib/dma.h:102]: Apologies @andyross , I was testing with all AMD/MTK/NXP/Intel targets in upstream, but right, definitions not used in upstream were very likely missed.

**Review by lgirdwood — COMMENTED** [review]
LGTM @andyross do you know if next step is to use native zephyr drivers for MTK ?

File: app/boards/acp_6_0_adsp.conf [hunk 6]
```diff
@@ -5,7 +5,6 @@ CONFIG_DCACHE_LINE_SIZE=128
 CONFIG_DYNAMIC_INTERRUPTS=y
```
Comment (kv2019i) [inline app/boards/acp_6_0_adsp.conf:5]: @andyross Ack. CONFIG_TRACE was working with Zephyr at some point, but basicly as Intel has moved to native logging, this code is likely to have bitrotted. For Intel, we are ok to remove it. It was useful at start, but now that there is plenty of infra available in Zephyr (and plenty of examples in SOF how to use it), I don't think it's the worth keeping around anymore. And if it doesn't work anymore, seems like a clear case.

File: src/math/Kconfig [hunk 7]
```diff
@@ -128,9 +128,7 @@ endmenu
 # this choice covers math iir, math fir, tdfb, and eqfir, eqiir.
 choice "FILTER_SIMD_LEVEL_SELECT"
 	prompt "choose which SIMD level used for IIR/FIR/TDFB module"
-	depends on COMP_FIR
-	depends on COMP_IIR
-	depends on COMP_TDFB
+	depends on COMP_FIR || COMP_IIR || COMP_TDFB
```
Comment (kv2019i) [inline src/math/Kconfig:131]: FYI @singalsu

File: src/lib/alloc.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (andyross) [reply src/lib/alloc.c:536]: Uh.... indeed.  Never mind.  :)

What happened here is I started work from a vendor tree which had stuffed in a bunch of debug code, including calls in the heap layer, and so it got pulled in semi-automatically.  And it didn't build, so I fixed it and moved on.  Then at some point I realized this was the legacy heap and removed it, completely forgetting about this patch.  Will remove.

**Review by kv2019i — CHANGES_REQUESTED** [review]
Thanks @andyross . 90% of the PR is completely fine and ready to go. The XTENSA_TOOLS_VERSION change seems a bit orthogonal to other fixes so maybe could go in as a separate PR.  Other than that, only one issue that affects NXP IMX configs (see inline comment), but rest looks nice!

This is also timely and good opportunity to discuss about our transition layers. I'll make a separate discussion page entry on that. I have some further cleanups in progress for our "transitory tools", btu I'm now a bit hesitent to push them through as they may cause trouble like you had with the DMA interface naming changes.

**Comment by andyross** [comment]
> LGTM @andyross do you know if next step is to use native zephyr drivers for MTK ?

That's the idea.  It's just the two drivers: dummy_dma and afe.  The former might have a solution from NXP already, but if not it's pretty simple as I read it.  The latter is half way there already: it sources all its config from DTS on the Zephyr side, but then translates it to the legacy config struct for use by the existing driver code.  Shouldn't be too hard.  But for sure the first submission is going to be ZEPHYR_NATIVE_DRIVERS=n and IPC3.

I'm hammering out all the existing workaround right now and hope to have the submission out today.

File: scripts/xtensa-build-zephyr.py (hunk 1 above)
Comment (andyross) [reply scripts/xtensa-build-zephyr.py:792]: That's fair.  Will resubmit in isolation and we can resume discussion there.

Comment (andyross) [reply scripts/xtensa-build-zephyr.py]: Resubmitted in #9736

File: zephyr/include/rtos/interrupt.h (hunk 4 above)
Comment (andyross) [reply zephyr/include/rtos/interrupt.h]: Fixed.  Indeed unless there was a special config I can't see, the whole expression was just a special case for IMX8M.  (Worth pointing out that that call is vestigial anyway, there's no point in tracking IRQ numbers in SOF code in a Zephyr build, the drivers have them already)

**Review by kv2019i — APPROVED** [review]
Thanks, looks good now!

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@wszypelt @lrudyX Jenkins all passing, good to merge ?

**Review by dbaluta — APPROVED** [review]

**Comment by wszypelt** [comment]
@lgirdwood good to merge :)

**Pull request merged**